### PR TITLE
Add `selfClosingTag` option to `vue/html-closing-bracket-newline`

### PR DESCRIPTION
**Pull request by mussinbenarbia**
Resolves https://github.com/vuejs/eslint-plugin-vue/issues/1794

The implementation follows @ota-meshi's [suggestion](https://github.com/vuejs/eslint-plugin-vue/issues/1794#issuecomment-1032417595) of adding a `selfClosingTag` option to the `html-closing-bracket-newline` rule.

It keeps compatibility by falling back to the previous behavior if the user does not specify the new option.

### CODE REVIEW

**Comment by mussinbenarbia** [comment]
I plan to update the docs once the implementation is confirmed/approved 🙇 Should I mark the PR as draft when I do this? Or should I let the maintainers do it?

File: lib/rules/html-closing-bracket-newline.js
Comment (FloEdelmann) [inline lib/rules/html-closing-bracket-newline.js:27]: Can you please add parameter types as a JSDoc comment?

File: lib/rules/html-closing-bracket-newline.js
Comment (FloEdelmann) [inline lib/rules/html-closing-bracket-newline.js:59]: This is not a valid schema. I think this should work fine:

```suggestion
          selfClosingTag: {
            type: 'object',
            properties: {
              singleline: { enum: ['always', 'never'] },
              multiline: { enum: ['always', 'never'] }
            },
            additionalProperties: false,
            minProperties: 1
          }
```

**Comment by FloEdelmann** [comment]
> Should I mark the PR as draft when I do this?

Yes please; it makes it clear that the PR is not yet intended to be complete yet.

File: lib/rules/html-closing-bracket-newline.js
Comment (mussinbenarbia) [reply lib/rules/html-closing-bracket-newline.js:59]: @FloEdelmann Thank you! Just one question, does `minProperties: 1` force users to specify at least one of the two `singleline` or `multiline` properties? If that's the case, should we instead allow people to not set any specific handling for selfClosingTag if they prefer?

File: lib/rules/html-closing-bracket-newline.js
Comment (FloEdelmann) [inline lib/rules/html-closing-bracket-newline.js:54]: No empty line after a function's JSDoc comment please:

```suggestion
 */
function getExpectedLineBreaks(node, options, type) {
```

File: tests/lib/rules/html-closing-bracket-newline.js
Comment (FloEdelmann) [inline tests/lib/rules/html-closing-bracket-newline.js:134]:
```suggestion
      code: `
        <template>
          <MyComp
            :foo="foo"
          />
          <MyComp :foo="foo" />
        </template>
      `,
      options: [
        {
          selfClosingTag: {
            singleline: 'never',
            multiline: 'always'
          }
        }
      ]
```

File: tests/lib/rules/html-closing-bracket-newline.js
Comment (FloEdelmann) [inline tests/lib/rules/html-closing-bracket-newline.js:150]:
```suggestion
      code: `
        <template>
          <MyComp :foo="foo"
          />
          <MyComp
            :foo="foo" />
        </template>
      `,
      options: [
        {
          selfClosingTag: {
            singleline: 'always',
            multiline: 'never'
          }
        }
      ]
```

**Review by FloEdelmann — COMMENTED** [review]
Looks good overall. Can you please improve the test cases with my suggestions and check if they still pass? Please also add the documentation for the next review :slightly_smiling_face:

**Comment by mussinbenarbia** [comment]
@FloEdelmann Oops, I re-requested review before finishing the docs. I'm on them right now

**Comment by mussinbenarbia** [comment]
@FloEdelmann Docs updated. I tried to follow the same structure as https://eslint.vuejs.org/rules/html-self-closing Let me know if that looks OK!

File: docs/rules/html-closing-bracket-newline.md
Comment (FloEdelmann) [inline docs/rules/html-closing-bracket-newline.md:109]:
```suggestion
### `"selfClosingTag": { "multiline": "always" }`
```

File: docs/rules/html-closing-bracket-newline.md
Comment (FloEdelmann) [inline docs/rules/html-closing-bracket-newline.md:86]:
```suggestion
- `"never"` ... disallow line breaks before the closing bracket.

If `selfClosingTag` is not specified, the `singleline` and `multiline` options are inherited for self-closing tags.

Plus, you can use [`vue/html-indent`](./html-indent.md) rule to enforce indent-level of the closing brackets.
```

**Review by FloEdelmann — APPROVED** [review]
Looks good to me now. Thanks for your contribution! :)

**Review by FloEdelmann — APPROVED** [review]

**Review by ota-meshi — APPROVED** [review]
LGTM! Thank you!

**Pull request merged**
